engine/verification/verifier/verifiers.go

```diff
@@ -0,0 +1,204 @@
+package verifier
+
+import (
+	"fmt"
+
+	"github.com/dgraph-io/badger/v2"
+	"github.com/rs/zerolog"
+	"github.com/rs/zerolog/log"
+
+	"github.com/onflow/flow-go/cmd/util/cmd/common"
+	"github.com/onflow/flow-go/engine/execution/computation"
+	"github.com/onflow/flow-go/fvm"
+	"github.com/onflow/flow-go/fvm/environment"
+	"github.com/onflow/flow-go/model/flow"
+	"github.com/onflow/flow-go/model/verification/convert"
+	"github.com/onflow/flow-go/module"
+	"github.com/onflow/flow-go/module/chunks"
+	"github.com/onflow/flow-go/module/metrics"
+	"github.com/onflow/flow-go/state/protocol"
+	"github.com/onflow/flow-go/storage"
+	storagepebble "github.com/onflow/flow-go/storage/pebble"
+)
+
+// VerifyLastKHeight verifies the last k sealed blocks by verifying all chunks in the results.
+// It assumes the latest sealed block has been executed, and the chunk data packs have not been
+// pruned.
+func VerifyLastKHeight(k uint64, chainID flow.ChainID, protocolDataDir string, chunkDataPackDir string) error {
+	db, storages, chunkDataPacks, state, verifier, err := initStorages(chainID, protocolDataDir, chunkDataPackDir)
+	if err != nil {
+		return fmt.Errorf("could not init storages: %w", err)
+	}
+	defer db.Close()
+
+	lastSealed, err := state.Sealed().Head()
+	if err != nil {
+		return fmt.Errorf("could not get last sealed height: %w", err)
+	}
+
+	root := state.Params().SealedRoot().Height
+	from := lastSealed.Height - k + 1
```

Comment:
Maybe check if `k` <= `lastSealed.Height` to prevent overflow since `from` is uint64?